tests/testthat/_snaps/deconstruct.md

```diff
@@ -0,0 +1,255 @@
+       6 BNA   Nashville Intl                      36.1  -86.7   599    -6 A     Amer~
+       7 BOS   General Edward Lawrence Logan Intl  42.4  -71.0    19    -5 A     Amer~
+       8 BTV   Burlington Intl                     44.5  -73.2   335    -5 A     Amer~
+       9 BUF   Buffalo Niagara Intl                42.9  -78.7   724    -5 A     Amer~
+      10 BUR   Bob Hope                            34.2 -118.    778    -8 A     Amer~
+      # ... with 76 more rows
+    Code
+      dm_nycflights13(cycle = TRUE)$airports
+    Output
+      # A tibble: 86 x 8
+      # Keys:     `faa` | 2 | 0
+         faa   name                                 lat    lon   alt    tz dst   tzone
+       * <chr> <chr>                              <dbl>  <dbl> <dbl> <dbl> <chr> <chr>
+       1 ALB   Albany Intl                         42.7  -73.8   285    -5 A     Amer~
+       2 ATL   Hartsfield Jackson Atlanta Intl     33.6  -84.4  1026    -5 A     Amer~
+       3 AUS   Austin Bergstrom Intl               30.2  -97.7   542    -6 A     Amer~
+       4 BDL   Bradley Intl                        41.9  -72.7   173    -5 A     Amer~
+       5 BHM   Birmingham Intl                     33.6  -86.8   644    -6 A     Amer~
+       6 BNA   Nashville Intl                      36.1  -86.7   599    -6 A     Amer~
+       7 BOS   General Edward Lawrence Logan Intl  42.4  -71.0    19    -5 A     Amer~
+       8 BTV   Burlington Intl                     44.5  -73.2   335    -5 A     Amer~
+       9 BUF   Buffalo Niagara Intl                42.9  -78.7   724    -5 A     Amer~
+      10 BUR   Bob Hope                            34.2 -118.    778    -8 A     Amer~
+      # ... with 76 more rows
+
+# `dm()` and `new_dm()` can handle a list of `dm_keyed_tbl` objects
+
+    Code
+      tbl_sum(dm_output$d1)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+---
+
+    Code
+      tbl_sum(dm_output$d2)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+---
+
+    Code
+      tbl_sum(new_dm_output$d1)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+---
+
+    Code
+      tbl_sum(new_dm_output$d2)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+# `dm()` and `new_dm()` can handle a mix of tables and `dm_keyed_tbl` objects
+
+    Code
+      tbl_sum(dm_output$d1)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+---
+
+    Code
+      tbl_sum(dm_output$d2)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+---
+
+    Code
+      tbl_sum(new_dm_output$d1)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+---
+
+    Code
+      tbl_sum(new_dm_output$d2)
+    Output
+               Keys 
+      "--- | 0 | 0" 
+
+# arrange for keyed tables produces expected output
+
+    Code
+      dm$airlines %>% arrange(desc(name))
+    Output
+      # A tibble: 15 x 2
+      # Keys:     `carrier` | 1 | 0
+         carrier name                       
+       * <chr>   <chr>                      
```

Comment:
The star indicates that the resulting data frame has row names. Why?

Reply:
These seem to be introduced while subsetting; I will give it a closer look:

``` r
library(dm, warn.conflicts = FALSE)

dm <- dm_nycflights13()
dm$airlines
#> # A tibble: 15 x 2
#> # Keys:     `carrier` | 1 | 0
#>    carrier name                       
#>  * <chr>   <chr>                      
#>  1 9E      Endeavor Air Inc.          
#>  2 AA      American Airlines Inc.     
#>  3 AS      Alaska Airlines Inc.       
#>  4 B6      JetBlue Airways            
#>  5 DL      Delta Air Lines Inc.       
#>  6 EV      ExpressJet Airlines Inc.   
#>  7 F9      Frontier Airlines Inc.     
#>  8 FL      AirTran Airways Corporation
#>  9 HA      Hawaiian Airlines Inc.     
#> 10 MQ      Envoy Air                  
#> 11 UA      United Air Lines Inc.      
#> 12 US      US Airways Inc.            
#> 13 VX      Virgin America             
#> 14 WN      Southwest Airlines Co.     
#> 15 YV      Mesa Airlines Inc.
```

<sup>Created on 2022-07-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Reply:
Minimal reprex shows that rownames are introduced while adding the additional class to the keyed table:

```r
  structure(
    x,
    dm_key_info = list(
      pk = pk,
      fks_in = fks_in,
      fks_out = fks_out,
      uuid = uuid
    ),
    class = unique(c("dm_keyed_tbl", class(x)))
  )
```

Reprex:

``` r
library(tibble)
x <- tibble("x" = 1)
has_rownames(x)
#> [1] FALSE

x <- structure(x, class = class(x))
has_rownames(x)
#> [1] TRUE
```

<sup>Created on 2022-07-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

I guess we can just remove the rownames in `new_keyed_tbl()` itself to fix this.

Reply:
Yeah, I remember `structure()` misbehaves here. We could use tools from rlang or vctrs to avoid introducing them, but any solution is fine for now.